src/main/java/ai/chalk/internal/arrow/FeatherProcessor.java

```diff
@@ -215,6 +223,7 @@ public static void powerWrite(BaseWriter writer, Object value) throws Exception
     public static byte[] inputsToArrowBytes(Map<String, List<?>> inputs) throws Exception {
         List<FieldVector> fieldVectors = new ArrayList<>();
         var uniformListLength = -1;
+        List<RootAllocator> allocators = new ArrayList<>();
```

Comment:
Would be careful here, you will leak all these allocators if an exception is thrown. We should create one RootAllocator w/ an overall limit and then use `newChildAllocator` if we want to further subdivide.